ctapipe/instrument/subarray.py

```diff
@@ -538,7 +547,7 @@ def to_hdf(self, h5file, overwrite=False, mode="a"):
                 meta["reference_itrs_z"] = itrs.z.to_value(u.m)
 
     @classmethod
-    def from_hdf(cls, path):
+    def from_hdf(cls, path, focal_length_choice="effective"):
```

Comment:
do we expect other possibilities than equivalent and effective in the future?
If not, a boolean `effective=True` might be less error prone than typing a string?

Reply:
Good point,  I will use an enum instead

Reply:
it's possible that in the future this is a calibration parameter that is measured in-situ as well. I know in HESS, for example, we store both the nominal and measured and have to correct for the difference, and for the HESS CT5 telescope, there is even an autofocus system that can dynamically change the focal length. So I agree that an ENUM is best here, in case we add other ways to obtain this information.